src/main/java/org/opensearch/neuralsearch/processor/explain/ExplanationDetails.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.explain;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.List;
+
+/**
+ * DTO class to store value and description for explain details.
+ * Used in {@link org.opensearch.neuralsearch.processor.NormalizationProcessorWorkflow} to normalize scores across shards.
+ */
+@Value
+@AllArgsConstructor
+public class ExplanationDetails {
+    int docId;
+    List<Pair<Float, String>> scoreDetails;
+
+    public ExplanationDetails(List<Pair<Float, String>> scoreDetails) {
+        this(-1, scoreDetails);
```

Comment:
Why hardcode this -1 value?

Reply:
this will match default docId in SearchHit https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/SearchHit.java#L170, we set it when we don't have access to real docId value

Reply:
Can you add this explanation at line 24? As by the look of it, it will very difficult to guess. Also please the code ref as well.

Reply:
yup, added